src/app/stores/wallet/reducer.ts

```diff
@@ -263,6 +264,21 @@ const walletReducer = (
         ...state,
         isUnlocked: action.isUnlocked,
       };
+    case RenameAccountKey:
+      return {
+        ...state,
+        accountsList: state.accountsList.map((account) =>
+          account.accountType === action.updatedAccount.accountType &&
+          account.id === action.updatedAccount.id
+            ? action.updatedAccount
+            : account,
+        ),
+        selectedAccount:
+          state.selectedAccount?.accountType === action.updatedAccount.accountType &&
+          state.selectedAccount?.id === action.updatedAccount.id
+            ? { ...state.selectedAccount, accountName: action.updatedAccount.accountName }
+            : state.selectedAccount,
+      };
```

Comment:
question: why not adding this logic in useWalletReducer?

Reply:
good call! updated this logic